### PR TITLE
Rename v1beta1 clients for test

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes

This commit renames the v1beta1 apiVersion of clients and adds the apiVersion to `./parse/yaml.go` for test.

/kind misc
cc @lbernick @XinruZhang
# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [n/a] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [n/a] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: test/parse/yaml.go [hunk 1]
```diff
@@ -84,6 +115,16 @@ kind: Pipeline
 	return &pipeline
 }
 
+// MustParseV1Pipeline takes YAML and parses it into a *v1beta1.Pipeline
+func MustParseV1Pipeline(t *testing.T, yaml string) *v1.Pipeline {
+	var pipeline v1.Pipeline
+	yaml = `apiVersion: tekton.dev/v1
```
Comment (XinruZhang) [inline test/parse/yaml.go:121]: This looks like a good candidate of a constant string value :)

File: test/parse/yaml.go [hunk 2]
```diff
@@ -84,6 +115,16 @@ kind: Pipeline
 	return &pipeline
 }
 
+// MustParseV1Pipeline takes YAML and parses it into a *v1beta1.Pipeline
```
Comment (XinruZhang) [inline test/parse/yaml.go:118]: v1

**Review by XinruZhang — APPROVED** [review]

**Comment by JeromeJu** [comment]
/retest 
The codes in CI are not updated.

**Comment by JeromeJu** [comment]
Needs to incorporate the changes at https://github.com/tektoncd/pipeline/commit/2d38f5fa840291395178422d34b36b1bc739e2a2

**Pull request merged**